subsys/bluetooth/host/adv.c

```diff
@@ -2283,3 +2284,47 @@ void bt_hci_le_scan_req_received(struct net_buf *buf)
 }
 #endif /* defined(CONFIG_BT_BROADCASTER) */
 #endif /* defined(CONFIG_BT_EXT_ADV) */
+
+#if defined(CONFIG_BT_CTLR_VS_SCAN_REQ_RX)
+#if defined(CONFIG_BT_BROADCASTER)
+
+static const struct bt_le_vs_cb *vs_cb;
+
+int bt_le_vs_scan_req_cb(const struct bt_le_vs_cb *cb)
+{
+	vs_cb = cb;
+	return 0;
+}
```

Comment:
Did you consider making the callbacks a slist similar to the existing scan callbacks?